### PR TITLE
fix(nextjs): Use backend isomorphic request in withClerkMiddleware

### PR DESCRIPTION
**Pull request by dimkl**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

### CODE REVIEW

**Review by anagstef — APPROVED** [review]
@dimkl  Looks good! We may need to make minor changes for tests to pass.

**Review by nikosdouvlis — APPROVED** [review]

File: packages/nextjs/src/server/withClerkMiddleware.ts [hunk 1]
```diff
@@ -38,32 +38,20 @@ export const withClerkMiddleware: WithClerkMiddleware = (...args: unknown[]) =>
...
-      forwardedHost: headers.get('x-forwarded-host') || undefined,
-      referrer: headers.get('referer') || undefined,
-      userAgent: headers.get('user-agent') || undefined,
-      proxyUrl,
       isSatellite,
       domain,
-      searchParams: new URL(req.url).searchParams,
       signInUrl,
+      proxyUrl,
+      request: req,
```
Comment (BRKalow) [inline packages/nextjs/src/server/withClerkMiddleware.ts:54]: ❓ for my own understanding, does this helper now pull the relevant data from headers/cookies via the request object?

Comment (anagstef) [reply packages/nextjs/src/server/withClerkMiddleware.ts:54]: @BRKalow Yes! Our `backend` package will handle the extraction of all the headers/cookies/searchParams needed. Here is where we do this inside the `authenticateRequest` in `backend` package:
https://github.com/clerkinc/javascript/blob/29c014262ceb326787359c83da665d825a2181fd/packages/backend/src/tokens/request.ts#L120

Comment (BRKalow) [reply packages/nextjs/src/server/withClerkMiddleware.ts:54]: Thanks for the pointer! 🙇

File: packages/nextjs/src/server/withClerkMiddleware.ts
Comment (BRKalow) [inline packages/nextjs/src/server/withClerkMiddleware.ts:1]: We can remove the now unused `getCookie` import from L18.

**Review by panteliselef — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**